src/fsharp/FSharp.Core.Unittests/FSharp.Core/Microsoft.FSharp.Control/AsyncType.fs

```diff
@@ -162,7 +162,22 @@ type AsyncType() =
         this.WaitASec t
         Assert.IsTrue (t.IsCompleted)
         Assert.AreEqual(s, t.Result)    
-      
+
+    [<Test>]
+    member this.RunSynchronouslyCancellationWithDelayedResult () =
```

Comment:
Could you add some kind of timeout to this test? Assuming the fix is not in place that test would run forever. Perhaps configurable via nunit?